### PR TITLE
Uncaught Errors, Cleanup, Styleguide, and Modals!

### PR DESCRIPTION
**Pull request by pbouzakis**

### CODE REVIEW

File: doc/coding-styleguide.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Coding Styleguide
+
+This is a work in progress. We are documenting the styleguide as we standardize or agree up on convention.
```
Comment (c-h-russell-walker) [inline doc/coding-styleguide.md]: `up on` should be `upon` ;-)

File: app/js/ui/CreateAccount.js [hunk 2]
```diff
@@ -42,7 +42,7 @@ export default class CreateAccount extends React.Component {
 
     @autobind @preventDefault
     _handleClick() {
-        console.log('create account!');
+        throw new Error("Not implemented: CreateAccount.");
```
Comment (c-h-russell-walker) [inline app/js/ui/CreateAccount.js:45]: I like this idea for development - throwing not implemented errors.

File: app/js/ui/SimpleModal.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        );
+    }
+
+    get _modalOpts() {
+        return { complete: this._handleModalComplete };
+    }
+
+    @autobind
+    _handleModalComplete() {
+        if (typeof this.props.onComplete === "function") {
```
Comment (c-h-russell-walker) [inline app/js/ui/SimpleModal.js:30]: I wonder if there's a cool way to re-use this pattern other places - would maybe make it more succinct as well.
`if (typeof this.props.onComplete === "function") {`

**Comment by c-h-russell-walker** [comment]
looks good to me!

File: app/js/ui/SimpleModal.js (hunk 3 above)
Comment (pbouzakis) [inline app/js/ui/SimpleModal.js:30]: Let's see when we need again, and we can make a decorator or mixin

**Pull request merged**